pandas-stubs/_typing.pyi

```diff
@@ -203,7 +204,20 @@ XMLParsers = Literal["lxml", "etree"]
 # Any plain Python or numpy function
 Function = Union[np.ufunc, Callable[..., Any]]
 GroupByObjectNonScalar = Union[
-    tuple, list[HashableT], Function, Series, np.ndarray, Mapping[Label, Any], Index
+    tuple,
+    list[HashableT],
+    Function,
+    list[Function],
+    Series,
+    list[Series],
+    np.ndarray,
+    list[np.ndarray],
+    Mapping[Label, Any],
+    list[Mapping[Label, Any]],
+    Index,
+    list[Index],
+    Grouper,
+    list[Grouper],
```

Comment:
If you are going to add all of these ones of the pattern `list[XXX]`, then create a test for each one.  e.g., for `list[Series]`, create an example that does a `groupby` on a list of series.

If all of those work, then also create an issue or PR in the pandas-repo that clarifies that those lists can be used as well.

Reply:
I did add the relevant tests to `test_series.py, test_frame.py`.
The pandas docs are already updated to include all these types, see here:
https://pandas.pydata.org/docs/dev/reference/api/pandas.DataFrame.groupby.html

Reply:
I think you don't have tests for `list[np.ndarray]`, `list[Mapping[Label, Any]]` and `list[Index]`